modules/nux/modules.php

```diff
@@ -260,47 +273,264 @@ public function process() {
+                            $value['user']['password'],
+                            false,
+                            null,
+                            false,
+                            'jmap',
+                            $this,
+                            false
+                        );
+                        Hm_Msgs::add("JMAP Server " . $key . " saved");
+                    }
+                    if ($value['smtp'] && !empty($value['smtp']['smtp_server'])) {
+                        // Check if module is supported
+                        if (!$this->module_is_supported('smtp')) {
+                            Hm_Msgs::add("ERRSMTP module is not enabled");
+                        } else {
+                            //TO DO: check first if the smtp_server server is already configured before connecting. can use the in_server_list function or create a new one specifically for my case
+                            $smtp_server_id = connect_to_smtp_server(
+                                $value['smtp']['smtp_server'],
+                                $key,
+                                $value['smtp']['smtp_port'],
+                                $value['user']['username'],
+                                $value['user']['password'],
+                                $value['smtp']['smtp_tls'],
+                                false
+                            );
+                            Hm_Msgs::add("SMTP Server " . $key . " saved");
+                        }
+                    }
+
+                    if ($value['imap'] && !empty($value['imap']['imap_server'])) {
+                        // Check if module is supported
+                        if (!$this->module_is_supported('imap')) {
+                            Hm_Msgs::add("ERRIMAP module is not enabled");
+                            return;
+                        }
+                        //TO DO: check first if the imap_server server is already configured before connecting can use the in_server_list function or create a new one specifically for my case
+                        $imap_server_id = connect_to_imap_server(
+                            $value['imap']['imap_server'],
+                            $key,
+                            $value['imap']['port'],
+                            $value['user']['username'],
+                            $value['user']['password'],
+                            $value['imap']['tls'],
+                            $value['sieve']['sieve_host'],
+                            $value['sieve']['sieve_port'],
+                            'imap',
+                            $this
+                        );
+                        Hm_Msgs::add("Server " . $key . "  saved");
+                    }
+                }
+
+                $this->save_hm_msgs();
+            }elseif(!is_null($file) && $file['type'] == 'text/csv'){
+                try {
+                    $file = new SplFileObject($file['tmp_name']);
```

Comment:
We can abstract this and make it a function which will return an array. Then other tests will be made here